source/MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.cpp

```diff
@@ -0,0 +1,111 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.h>
+#include <MaterialXRuntime/Private/PvtCommand.h>
+
+#include <MaterialXRuntime/RtPrim.h>
+#include <MaterialXRuntime/RtTypeDef.h>
+#include <MaterialXRuntime/RtValue.h>
+
+namespace MaterialX
+{
+
+PvtRemoveMetadataCmd::PvtRemoveMetadataCmd(RtObject& obj, const RtToken& name)
+    : PvtCommand()
+    , _obj(obj)
+    , _name(name)
+    , _oldValue()
+{
+}
+
+PvtCommandPtr PvtRemoveMetadataCmd::create(RtObject& obj, const RtToken& name)
+{
+    return std::make_shared<PvtRemoveMetadataCmd>(obj, name);
+}
+
+void PvtRemoveMetadataCmd::execute(RtCommandResult& result)
+{
+    if (_obj.isValid())
+    {
+        try
+        {
+            // Send message that the metadata is being removed
+            msg().sendRemoveMetadataMessage(_obj, _name);
+
+            // Save old value for undo/redo
+            _oldValue = RtValue::clone(RtType::STRING, _obj.getMetadata(_name)->getValue(), _obj.getParent());
+
+            // Remove the metadata value
+            _obj.removeMetadata(_name);
+            result = RtCommandResult(true);
+        }
+        catch (const ExceptionRuntimeError& e)
+        {
+            result = RtCommandResult(false, string(e.what()));
+        }
+    }
+    else
+    {
+        result = RtCommandResult(false, string("Node to remove metadata from is no longer valid"));
+    }
+}
+
+void PvtRemoveMetadataCmd::undo(RtCommandResult& result)
+{
+    if (_obj.isValid())
+    {
+        try
+        {
+            // Send message that the metadata is being created
+            msg().sendCreateMetadataMessage(_obj, _name, _oldValue);
+
+            RtTypedValue* md = _obj.addMetadata(_name, RtType::STRING);
+            if (md)
```

Comment:
This test should not be needed, addMetadata cannot return nullptr. 

Unless the unlikely event we run totally out of memory (both physical and virtual memory) but that is a fatal error we can't repair anyway, and we don't handled anywhere else, so we would crash and burn nonetheless if that happens.